pkg/openservicebroker/validation.go

```diff
@@ -0,0 +1,47 @@
+package openservicebroker
+
+import (
+	"fmt"
+	"regexp"
+
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	"github.com/openshift/origin/pkg/template/api/validation"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+)
+
+func ValidateProvisionRequest(preq *ProvisionRequest) field.ErrorList {
+	errors := ValidateUUID(field.NewPath("service_id"), preq.ServiceID)
+	errors = append(errors, ValidateUUID(field.NewPath("plan_id"), preq.PlanID)...)
+
+	for key := range preq.Parameters {
+		if !validation.ParameterNameRegexp.MatchString(key) &&
+			key != templateapi.NamespaceParameterKey &&
+			key != templateapi.ImpersonateUserParameterKey {
+			errors = append(errors, field.Invalid(field.NewPath("parameters."+key), key, fmt.Sprintf("does not match %v", validation.ParameterNameRegexp)))
+		}
+	}
+
+	return errors
+}
+
+func ValidateBindRequest(breq *BindRequest) field.ErrorList {
+	errors := ValidateUUID(field.NewPath("service_id"), breq.ServiceID)
+	errors = append(errors, ValidateUUID(field.NewPath("plan_id"), breq.PlanID)...)
+
+	for key := range breq.Parameters {
+		if !validation.ParameterNameRegexp.MatchString(key) {
+			errors = append(errors, field.Invalid(field.NewPath("parameters."+key), key, fmt.Sprintf("does not match %v", validation.ParameterNameRegexp)))
+		}
```

Comment:
what parameters do we expect on a bind?  what would we do with them?

perhaps it's an error if we get any parameters?  either that or we should just ignore any parameters that are provided.